### ISSUE TITLE
-Wunguarded-availability can't cope with early returns

### ISSUE THREAD

**Issue by DanAlbert** [llvm/llvm-project#93107]
https://godbolt.org/z/s1YGsPW4s

```c++
#include <stdlib.h>

extern "C" void foo() __attribute__((availability(android, introduced=30)));

int main() {
    // This weird pattern is required:
    // https://github.com/llvm/llvm-project/issues/33161
    if (__builtin_available(android 30, *)) {
    } else {
        return 1;
    }

    foo();
}
```

When compiled with `-Werror=unguarded-availability -target aarch64-linux-android21 foo.cpp`:

```
<source>:13:5: error: 'foo' is only available on Android 30 or newer [-Werror,-Wunguarded-availability]
   13 |     foo();
      |     ^~~
<source>:3:17: note: 'foo' has been marked as being introduced in Android 30 here, but the deployment target is Android 21
    3 | extern "C" void foo() __attribute__((availability(android, introduced=30)));
      |                 ^
<source>:13:5: note: enclose 'foo' in a __builtin_available check to silence this warning
   13 |     foo();
      |     ^~~   
1 error generated.
Compiler returned: 1
```
